api/src/helper/lib/base-nlp-helper.ts

```diff
@@ -226,6 +226,11 @@ export default abstract class BaseNlpHelper<
     project?: string,
   ): Promise<NLU.ParseEntities>;
 
+  private buildUnicodeRegexExpression(term: string): RegExp {
+    const escapedTerm = term.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
Hi @MohamedAliBouhaouala, you can use the **escapeRegExp** function.
The mentioned function is already used in `api/src/utils/pipes/search-filter.pipe.ts`
<img width="452" height="42" alt="image" src="https://github.com/user-attachments/assets/cf7be58a-0549-4637-8f35-2561716b6190" />